src/LendingPool.sol

```diff
@@ -18,19 +18,19 @@ import "./interfaces/IVault.sol";
 import "./interfaces/ILendingPool.sol";
 import {TrustedProtocol} from "./TrustedProtocol.sol";
 import {DebtToken} from "./DebtToken.sol";
+import {InterestRateModule} from "./libraries/InterestRateModule.sol";
 
 /**
  * @title Lending Pool
  * @author Arcadia Finance
  * @notice The Lending pool contains the main logic to provide liquidity and take or repay loans for a certain asset
  */
-contract LendingPool is Owned, TrustedProtocol, DebtToken {
+contract LendingPool is Owned, TrustedProtocol, DebtToken, InterestRateModule {
     using SafeTransferLib for ERC20;
     using FixedPointMathLib for uint256;
 
-    uint256 public constant YEARLY_BLOCKS = 2628000;
+    uint256 public constant YEARLY_BLOCKS = 2_628_000;
```

Comment:
line 43: `uint256[] public weights;`

this could be a uint16/24, allows it to pack the values in one slot. Since they are mostly all read, will save quite some gas

Reply:
I am creating task for this, It can be bundled with other improvement tasks

Reply:
https://www.notion.so/arcadiafinance/Pack-values-into-single-var-09d1fb8461af4f1ab6b4bfef414cf21b